lib/rspec-puppet/support.rb

```diff
@@ -217,7 +217,23 @@ def with_vardir
       end
     end
 
+    def reset_providers
+      Puppet::Type.eachtype do |type|
+        # Unset the default provider for each type as this is cached in a class
+        # instance method and so doesn't get re-evaluated when we change facts
+        # and compile a new catalogue.
+        type.defaultprovider = nil
+      end
+    end
+
     def build_catalog_without_cache(nodename, facts_val, trusted_facts_val, hiera_config_val, code, exported)
+      # Blindly stub out any feature and command Puppet::Confine evaluations
+      # that providers might do. This is necessary because we convert the
+      # resource catalogue to a RAL catalogue in order to evaluate all the
+      # automatic graph edges, which requires evaluating the providers.
+      allow_any_instance_of(Puppet::Confine::Feature).to receive(:pass?).with(anything).and_return(true)
+      allow_any_instance_of(Puppet::Util).to receive(:which).with(anything).and_return(true)
+      allow_any_instance_of(Puppet::Provider::Command).to receive(:execute).with(any_args).and_return("")
```

Comment:
I don't know how provider selection and validation happens on the server. In the end, you're the expert here, and I trust your judgement. In the worst case, we have to revisit this in a few weeks when we get windows testing online.